DomainContent/Mexico/onPinataHit.js

```diff
@@ -0,0 +1,182 @@
+//
+// onPinataHit.js
+// A script to handle pinata behavior
+// 
+// Author: Elisa Lupin-Jimenez
+// Copyright High Fidelity 2018
+//
+// Licensed under the Apache 2.0 License
+// See accompanying license file or http://apache.org/
+//
+// All assets are under CC Attribution Non-Commerical
+// http://creativecommons.org/licenses/
+//
+
+(function() {
+
+    var WHACK_SOUND;
+    var PARTY_SOUND;
+    var paddleID;
+    var paddleOriginalPosition;
+    var pinataID;
+    var pinataOriginalPosition;
+
+    this.preload = function(entityID) {
+        paddleID = entityID;
+        paddleOriginalPosition = Entities.getEntityProperties(entityID, "position").position;
+        var pinataArray = Entities.findEntities(paddleOriginalPosition, 3);
+        for (i in pinataArray) {
+            var objectID = pinataArray[i];
+            if (Entities.getEntityProperties(objectID, "name").name === "Pinata") {
+                pinataID = objectID;
+                pinataOriginalPosition = Entities.getEntityProperties(objectID, "position").position;
+            }
+        }
+        // sound from http://soundbible.com/1952-Punch-Or-Whack.html
+        WHACK_SOUND = SoundCache.getSound("https://hifi-content.s3.amazonaws.com/elisalj/mexico/whack.wav");
+        PARTY_SOUND = SoundCache.getSound("http://mpassets.highfidelity.com/a5f42695-f15a-4f44-9660-14b4f8ca2b29-v1/PartyHorn4.wav");
+    };
+
+    var hits = 0;
+    var hitsToBreak = Math.random() * 10;
+    var TIMEOUT = 30000;
+    var CANDY_LIFETIME = 30;
+
+    var PARTICLE_COUNT = 8;
+
+    var CONFETTI = {
+        type: "ParticleEffect",
+        isEmitting: 1,
+        maxParticles: 8,
+        lifespan: 10,
+        lifetime: 3,
+        emitRate: 100,
+        emitSpeed: 3,
+        speedSpread: 1,
+        script: "(function() { return { preload: function(e) { Script.setTimeout(function(){Entities.editEntity(e, { 'isEmitting': 0, 'script':'' }) },50); } } })",
```

Comment:
This should probably not be an inline script. If we have to put an entity script filter on the domain, it will break this functionality.

Reply:
I borrowed this from the confetti gun, when I took this off, the particles did not clear quickly enough.